### DISCUSSION TITLE
Is there a difference between using multiple taffy instances vs one for multiple root-nodes?

### DISCUSSION THREAD

**Discussion by hasezoey** [DioxusLabs/taffy#524] [Q&A]
I am somewhat curious as to if there is a difference between using one `Taffy` instance vs multiple, where there are multiple trees (root-nodes) which are unrelated to each-other.

for example you have 2 components, which each have a root-node and are not related to each-other, should they each have their own taffy instances or should one "more global" one be used, what are the differences (if any)?

**Comment by nicoburns** [comment]
There's not a big difference at the moment, no. Having everything in one will require fewer allocations (but might use slightly more memory) due to everything being stored in one set of slotmaps. Having separate instances is arguably a bit cleaner from a code perspective.
